packages/mui-material/src/ComponentWithComponentProp.d.ts

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import { CommonProps } from './OverridableComponent';
+
+export type Component<Props = any> = React.ElementType<Props>;
```

Comment:
I wonder if we even need this type - it's merely a rename of React type and introduces an abstraction to keep in mind.